src/rulesets/Base/mapreduce.jl

```diff
@@ -6,7 +6,21 @@ function frule((_, ẋ), ::typeof(sum), x; dims=:)
     return sum(x; dims=dims), sum(ẋ; dims=dims)
 end
 
-function rrule(::typeof(sum), x::AbstractArray{T}; dims=:) where {T<:Number}
+function sum_rrule(x, dims::Colon)
+    y = sum(x; dims=dims)
+    let xdims=size(x)
+        function sum_pullback(ȳ)
+            x̄ = InplaceableThunk(
+                x -> x .+= ȳ,
+                @thunk(fill(ȳ, xdims...)),
```

Comment:
Can I suggest using `axes`?
```suggestion
    y = sum(x)
    let x_axes = axes(x)
        function sum_pullback(ȳ)
            x̄ = InplaceableThunk(
                x -> x .+= ȳ,
                @thunk(fill(ȳ, x_axes)),
```